api.js

```diff
@@ -177,24 +187,30 @@ Api.prototype.run = function (files) {
 			var tests = files.map(self._runFile);
 
 			// receive test count from all files and then run the tests
+			var runOnlyExclusive = false;
 			var statsCount = 0;
 
 			return new Promise(function (resolve) {
 				tests.forEach(function (test) {
 					var counted = false;
 
-					function tryRun() {
+					function tryRun(stats) {
 						if (counted) {
 							return;
 						}
 
+						runOnlyExclusive = runOnlyExclusive || stats.hasExclusive;
```

Comment:
I like this approach! Seems cleaner than the `this.hasExclusive` approach on top, though it can't fix the `testCount`. Maybe the test count should be incremented here as well and we can do away with `_handleStats`?

Note though that `tryRun` is also called when on `test.catch()`. You'd have to wrap it there to avoid passing the error as the `stats` object and then add a guard for `stats` being undefined.

Or maybe stick with the `_handleStats()` approach. Just not both :wink: